x/reporter/keeper/distribution.go

```diff
@@ -27,49 +25,41 @@ import (
 // 5. Distributes the net reward among the selectors based on their shares.
 // 6. Adds the commission to the reporter's share if the reporter is also a selector.
 // 7. Updates the selectors' tips with the new calculated shares.
-func (k Keeper) DivvyingTips(ctx context.Context, reporterAddr sdk.AccAddress, reward types.BigUint, queryId []byte, height uint64) error {
+func (k Keeper) DivvyingTips(ctx context.Context, reporterAddr sdk.AccAddress, reward math.LegacyDec, queryId []byte, height uint64) error {
 	reporter, err := k.Reporters.Get(ctx, reporterAddr)
 	if err != nil {
 		return err
 	}
 
-	// Convert arguments needed for calculations to legacy decimals
-	rewardDec := k.LegacyDecFromMathUint(reward.Value)
-	commissionRateDec := k.LegacyDecFromMathUint(reporter.CommissionRate)
-
-	// Calculate commission: commission = reward * commissionRate
-	commissionDec := rewardDec.Mul(commissionRateDec).Quo(math.LegacyNewDec(1000000))
-
-	commission := k.TruncateUint(commissionDec)
-
+	// selector's commission = reporter's commission rate * reward
+	commission := reward.Mul(reporter.CommissionRate)
```

Comment:
Should this be truncated before using it again ?

Reply:
if you're asking because of consistency, maybe this shouldn't be truncated, `delegatorShare = delegatorShare.TruncateDec()`,